### PR TITLE
CSHARP-2942: Check uses of GetCustomAttributes when targetting netstandard2.0

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
JIRA: https://jira.mongodb.org/browse/CSHARP-2942
EG: https://evergreen.mongodb.com/version/607f283530661503dcfb078f

### CODE REVIEW

File: src/MongoDB.Bson/Serialization/BsonSerializer.cs [hunk 1]
```diff
@@ -692,12 +692,8 @@ internal static void EnsureKnownTypesAreRegistered(Type nominalType)
...
                 {
                     // only call LookupClassMap for classes with a BsonKnownTypesAttribute
-#if NET452
-                    var knownTypesAttribute = nominalType.GetTypeInfo().GetCustomAttributes(typeof(BsonKnownTypesAttribute), false);
-#else
-                    var knownTypesAttribute = nominalType.GetTypeInfo().GetCustomAttributes(typeof(BsonKnownTypesAttribute), false).ToArray();
-#endif
-                    if (knownTypesAttribute != null && knownTypesAttribute.Length > 0)
+                    var hasAttribute = nominalType.GetTypeInfo().GetCustomAttributes(inherit: false).OfType<BsonKnownTypesAttribute>().Any();
+                    if (hasAttribute)
```
[the remark this answers is not in the document]
Comment (rstam) [reply src/MongoDB.Bson/Serialization/BsonSerializer.cs:696]: Consider slightly more efficient:

    var hasKnownTypesAttribute = nominalType.GetTypeInfo().GetCustomAttributes(typeof(BsonKnownTypesAttribute), inherit: false).Any();

**Review by rstam — CHANGES_REQUESTED** [review]

File: src/MongoDB.Bson/Serialization/BsonSerializer.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikalaiMazurenka) [reply src/MongoDB.Bson/Serialization/BsonSerializer.cs]: Done

**Review by MikalaiMazurenka — COMMENTED** [review]
Ready for review!

**Review by rstam — APPROVED** [review]
LGTM

**Pull request merged**